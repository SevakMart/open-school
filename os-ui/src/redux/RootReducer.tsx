import { combineReducers } from 'redux';
import chooseSubcategoryReducer from './Slices/ChoosSubcategorySlice';
import userInfoReducer from './Slices/loginUserSlice';
import allLearningPathFilterParamsReducer from './Slices/AllLearningPathFilterParamsSlice';
import forgotPasswordEmailReducer from './Slices/ForgotPasswordEmailSlice';
import signedUpUserIdReducer from './Slices/SignedUpUserIdSlice';
import suggestedCourseReducer from './Slices/SuggestedCourseSlice';
import userEnrolledCourseByCourseStatusReducer from './Slices/UserEnrolledCourseByCourseStatus';
import allLearningPathCoursesReducer from './Slices/AllLearningPathCourseSlice';
import allMentorsFilterParamsReducer from './Slices/AllMentorsFilterParamsSlice';
import allMentorsSliceReducer from './Slices/AllMentorsSlice';
import savedMentorsReducer from './Slices/SavedMentorsSlice';
import savedCourseReducer from './Slices/SavedLearningPathSlice';
import homepageMentorListReducer from './Slices/HomepageMentorSlice';
import portalStatusReducer from './Slices/PortalOpenStatus';

const rootReducer = combineReducers({
  portalStatus: portalStatusReducer,
  userInfo: userInfoReducer,
  chooseSubcategories: chooseSubcategoryReducer,
  filterParams: allLearningPathFilterParamsReducer,
  forgotPasswordEmail: forgotPasswordEmailReducer,
  signedUpUserId: signedUpUserIdReducer,
  suggestedCourse: suggestedCourseReducer,
  userEnrolledCourseByCourseStatus: userEnrolledCourseByCourseStatusReducer,
  allLearningPathCourses: allLearningPathCoursesReducer,
  allMentorsFilterParams: allMentorsFilterParamsReducer,
  allMentorsList: allMentorsSliceReducer,
  savedMentors: savedMentorsReducer,
  savedCourse: savedCourseReducer,
  homepageMentorList: homepageMentorListReducer,
});
export default rootReducer;
