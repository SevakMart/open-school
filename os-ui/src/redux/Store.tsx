import { configureStore } from '@reduxjs/toolkit';
import {
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from 'redux-persist';
import { createStateSyncMiddleware, initMessageListener } from 'redux-state-sync';
import rootReducer from './RootReducer';

export const store = configureStore({
  reducer: rootReducer,
  middleware: (getDefaultMiddleware) => getDefaultMiddleware({
    serializableCheck: {
      ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
    },
  }).concat(
    createStateSyncMiddleware({
      blacklist: ['persist/PERSIST', 'persist/REHYDRATE'],
    }),
  ),
});

initMessageListener(store);

export type RootState = ReturnType<typeof store.getState>
