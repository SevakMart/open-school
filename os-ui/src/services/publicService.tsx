import { fetchDataGet } from './fetchData';

class PublicService {
  readonly baseUrl: string;

  constructor() {
    this.baseUrl = '/api/v1/public';
  }

  async getPublicCategories(params: object = {}) {
    const data = await fetchDataGet(`${this.baseUrl}/categories`, params);
    return data;
  }

  async getPublicMentors(params:object = {}) {
    const data = await fetchDataGet(`${this.baseUrl}/users/mentors`, params);
    return data;
  }
}

export default new PublicService();
