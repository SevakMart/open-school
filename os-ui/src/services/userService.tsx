import fetchService from './fetchData';

class UserService {
  readonly basePath: string;

  constructor() {
    this.basePath = 'users';
  }

  async getUserCourses(userId:number, userToken:string, params = {}) {
    const response = await fetchService.get(`${this.basePath}/${userId}/courses/enrolled`, params, userToken);
    const { status } = response;
    const data = await response.json();
    return { data, status };
  }

  async getSuggestedCourses(userId:number, userToken:string) {
    const response = await fetchService.get(`${this.basePath}/${userId}/courses/suggested`, {}, userToken);
    const { status } = response;
    const data = await response.json();
    return { data, status };
  }

  async getMentors(params:object = {}, token:string) {
    const response = await fetchService.get(`${this.basePath}/mentors`, params, token);
    const { status } = response;
    const data = await response.json();
    return { data, status };
  }

  async savePreferredCategories(userId:number, userToken:string, content:Array<number>) {
    const data = await (await fetchService.post(`${this.basePath}/${userId}/categories`, content, {}, userToken)).json();
    return data;
  }

  async saveUserPreferredCourses(userId:number, courseId:number, userToken:string) {
    const data = await (await fetchService.post(`${this.basePath}/${userId}/courses/saved`, { courseId }, {}, userToken)).json();
    return data;
  }

  async getUserSavedCourses(userId:number, userToken:string, params = {}) {
    const data = await (await fetchService.get(`${this.basePath}/${userId}/courses/saved`, params, userToken)).json();
    return data;
  }

  async deleteUserSavedCourses(userId:number, courseId:number, userToken:string, params = {}) {
    const response = await fetchService.delete(`${this.basePath}/${userId}/courses/${courseId}/saved`, params, userToken);
    const { status } = response;
    const data = await response.json();
    return { data, status };
  }

  async saveUserMentor(userId:number, mentorId:number, userToken:string, params = {}) {
    const data = await (await fetchService.post(`${this.basePath}/${userId}/mentors/${mentorId}`, null, params, userToken)).json();
    return data;
  }

  async deleteUserSavedMentor(userId:number, mentorId:number, userToken:string, params = {}) {
    const response = await fetchService.delete(`${this.basePath}/${userId}/mentors/${mentorId}/saved`, params, userToken);
    const { status } = response;
    const data = await response.json();
    return { data, status };
  }
}
export default new UserService();
