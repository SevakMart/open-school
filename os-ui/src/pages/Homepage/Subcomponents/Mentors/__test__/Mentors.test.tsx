import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { store } from '../../../../../redux/Store';
import publicService from '../../../../../services/publicService';
import userService from '../../../../../services/userService';
import HomepageMentors from '../Mentors';

/* const notSignInData = {
  data: {
    content: [
      {
        name: 'John',
        surname: 'Smith',
        professionName: 'JS Developer',
        companyName: 'Google',
        courseCount: 5,
        userImgPath: 'https://image.shutterstock.com/image-vector/businessman-avatar-profile-picture-260nw-221565274.jpg',
        emailPath: 'fakeEmail.com',
        linkedinPath: 'https://linkedin.com/feed',
      },
      {
        name: 'Jane',
        surname: 'Smith',
        professionName: 'React Developer',
        companyName: 'Facebook',
        courseCount: 16,
        userImgPath: 'https://image.shutterstock.com/image-vector/businessman-avatar-profile-picture-260nw-221565274.jpg',
        emailPath: 'fakeEmail.com',
        linkedinPath: 'https://linkedin.com/feed',
      },
    ],
    totalPages: 2,
  },
  status: 200,
};
const signInData = {
  data: {
    content: [
      {
        name: 'Jake',
        surname: 'Smith',
        professionName: 'C# Developer',
        companyName: 'Microsoft',
        courseCount: 10,
        userImgPath: 'https://image.shutterstock.com/image-vector/businessman-avatar-profile-picture-260nw-221565274.jpg',
        emailPath: 'fakeEmail.com',
        linkedinPath: 'https://linkedin.com/feed',
      },
      {
        name: 'Mark',
        surname: 'Smith',
        professionName: 'NodeJS Developer',
        companyName: 'Amazon',
        courseCount: 7,
        userImgPath: 'https://image.shutterstock.com/image-vector/businessman-avatar-profile-picture-260nw-221565274.jpg',
        emailPath: 'fakeEmail.com',
        linkedinPath: 'https://linkedin.com/feed',
      },
    ],
    totalPages: 2,
  },
  status: 200,
};

describe('Create test cases for homepage mentors list', () => {
  test('Create a snapshot test', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <HomepageMentors handleButtonClick={() => null} />
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
  test('Check if the right data is displayed when not signed in', async () => {
    expect.hasAssertions();
    jest.spyOn(publicService, 'getPublicMentors').mockResolvedValue(notSignInData);
    render(
      <Provider store={store}>
        <HomepageMentors handleButtonClick={() => null} />
      </Provider>,
    );
    const mentorName = await screen.findByTestId('John Smith');
    expect(mentorName).toBeInTheDocument();
  });
  test('Check if the right data is displayed when signed in', async () => {
    expect.hasAssertions();
    jest.spyOn(userService, 'getMentors').mockResolvedValue(signInData);
    render(
      <Provider store={store}>
        <HomepageMentors handleButtonClick={() => null} />
      </Provider>,
    );
    const mentorName = await screen.findByTestId('Mark Smith');
    expect(mentorName).toBeInTheDocument();
  });
  test('Test fetching empty mentor list', async () => {
    expect.hasAssertions();
    jest.spyOn(publicService, 'getPublicMentors').mockResolvedValue({ data: { content: [], totalPages: 0 }, status: 200 });
    render(
      <Provider store={store}>
        <HomepageMentors handleButtonClick={() => null} />
      </Provider>,
    );
    const emptyCategoryHeading = await screen.findByTestId('emptyMentorMessage');
    expect(emptyCategoryHeading).toBeInTheDocument();
  });
});
*/
