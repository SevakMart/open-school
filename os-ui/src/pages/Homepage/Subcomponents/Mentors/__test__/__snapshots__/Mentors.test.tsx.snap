// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create test cases for homepage mentors list Create a snapshot test 1`] = `
<DocumentFragment>
  <div
    class="mentorMainContainer"
  >
    <div
      class="main"
      style="padding-top: 5%;"
    >
      <h3>
        string.homePage.mentors.ourMentors
      </h3>
      <h2>
        string.homePage.mentors.learnAboutContributors
      </h2>
    </div>
    <div
      class="listContainer"
      style="margin: 6%;"
    >
      <div
        class="gridContent"
      >
        <div
          class="mainContent"
        >
          <div
            class="loader"
          >
            <span />
            <span />
            <span />
          </div>
        </div>
      </div>
    </div>
    <div
      class="registrationButton"
      style="margin: auto; padding: 5% 0px 10%;"
    >
      <button
        class="mainMentorRegistrationButton"
        type="button"
      >
        button.homePage.registerMentor
      </button>
    </div>
  </div>
</DocumentFragment>
`;
